specification/text_structure/find_structure/FindStructureRequest.ts

```diff
@@ -0,0 +1,71 @@
+    format?: string
+    /** If you have set format to semi_structured_text, you can specify a Grok pattern that is used to extract fields from every message in the text. The name of the timestamp field in the Grok pattern must match what is specified in the timestamp_field parameter. If that parameter is not specified, the name of the timestamp field in the Grok pattern must match "timestamp". If grok_pattern is not specified, the structure finder creates a Grok pattern. */
+    grok_pattern?: string
+    /** If you have set format to delimited, you can use this parameter to indicate whether the column names are in the first row of the text. If this parameter is not specified, the structure finder guesses based on the similarity of the first row of the text to other rows. */
+    has_header_row?: boolean
+    /**
+     * The maximum number of characters in a message when lines are merged to form messages while analyzing semi-structured text. If you have extremely long messages you may need to increase this, but be aware that this may lead to very long processing times if the way to group lines into messages is misdetected.
+     * @server_default 10000
+     */
+    line_merge_size_limit?: uint
+    /**
+     * The number of lines to include in the structural analysis, starting from the beginning of the text. The minimum is 2; If the value of this parameter is greater than the number of lines in the text, the analysis proceeds (as long as there are at least two lines in the text) for all of the lines.
+     * @server_default 1000
+     */
+    lines_to_sample?: uint
+    /** If you have set format to delimited, you can specify the character used to quote the values in each row if they contain newlines or the delimiter character. Only a single character is supported. If this parameter is not specified, the default value is a double quote ("). If your delimited text format does not use quoting, a workaround is to set this argument to a character that does not appear anywhere in the sample. */
+    quote?: string
+    /** If you have set format to delimited, you can specify whether values between delimiters should have whitespace trimmed from them. If this parameter is not specified and the delimiter is pipe (|), the default value is true. Otherwise, the default value is false. */
+    should_trim_fields?: boolean
+    /**
+     * Sets the maximum amount of time that the structure analysis make take. If the analysis is still running when the timeout expires then it will be aborted.
+     * @server_default 25s
+     */
+    timeout?: Time
+    timestamp_field?: Field
+    /** The Java time format of the timestamp field in the text. */
+    timestamp_format?: string
+  }
+  body: TBody
```

Comment:
This is a ndjson api:
```suggestion
  body: Array<TBody>
```